src/router.ts

```diff
@@ -491,7 +556,9 @@ export class Router {
       let first = results[0];
       let instruction = new NavigationInstruction(Object.assign({}, instructionInit, {
         params: first.params,
-        queryParams: first.queryParams || results.queryParams,
+        queryParams: first.queryParams
+          // How is this valid
+          || (results as any).queryParams,
```

Comment:
It's kind of a funny class. It doesn't inherit from Array but it adopts some of the Array methods right off the prototype so it looks and feels like an array but isn't.

Assuming this is typed correctly above, you shouldn't need the `as any`

https://github.com/aurelia/route-recognizer/blob/master/src/route-recognizer.js#L252

Reply:
it is mistyped in route recognizer. do (as RecognizeResults) for now.